model/telemetry/registry.yaml

```diff
@@ -81,3 +81,57 @@ groups:
         brief: >
           The version string of the auto instrumentation agent or distribution, if used.
         examples: ["1.2.3"]
+      - id: telemetry.sdk.processor.type
+        type:
+          members:
+            - id: batch_span
+              value: "batch-span"
+              brief: >
+                The builtin SDK Batching Span Processor
+              stability: experimental
+            - id: simple_span
+              value: "simple-span"
+              brief: >
+                The builtin SDK Simple Span Processor
+              stability: experimental
+        stability: experimental
+        brief: >
+          A name identifying the type of the OpenTelemetry SDK processor.
+        examples: ["batch-span", "MyCustomProcessor"]
+      - id: telemetry.sdk.exporter.type
+        type:
+          members:
+            - id: otlp_grpc
+              value: "otlp-grpc"
+              brief: >
+                OTLP exporter over gRPC with protobuf serialization
+              stability: experimental
+            - id: otlp_http
+              value: "otlp-http"
+              brief: >
+                OTLP exporter over HTTP with protobuf serialization
+              stability: experimental
+            - id: otlp_http_json
+              value: "otlp-http-json"
+              brief: >
+                OTLP exporter over HTTP with JSON serialization
+              stability: experimental
+        stability: experimental
+        brief: >
+          A name identifying the type of the OpenTelemetry SDK exporter.
+        examples: ["otlp-grpc", "jaeger"]
+      - id: telemetry.sdk.component.id
+        type: string
+        stability: experimental
+        brief: >
+          A name uniquely identifying the instance of the OpenTelemetry SDK component within its containing SDK instance.
+        note: |
+          The SDK MAY allow users to provide an id for the component instances. If no id is provided by the user,
+          the SDK SHOULD automatically assign an id. Because this attribute is used in metrics, the SDK MUST ensure a low cardinality in that case.
```

Comment:
> Because this attribute is used in metrics, the SDK MUST ensure a low cardinality in that case.

How would SDK ensure it?

If we want to capture an index of the processing/exporting pipeline, perhaps we can add an index as a separate attribute and ask SDKs to set it? 

Also how would SDK deal with cases like

```logs
sdk -> custom_composite_processor1 ----> another_processor1 ---> exporter1
                                                              ----> processor2 ---> exporter2
```

it'd only know about `custom_composite_processor1` and the rest is opaque.

I think it's worth documenting that components are responsible for capturing their telemetry and disambiguating it across different instances of the same component.

Reply:
> If we want to capture an index of the processing/exporting pipeline, perhaps we can add an index as a separate attribute and ask SDKs to set it?

I'd say that is a non-goal for this attribute: Due to composition / wrapping, it would be impossible for the SDK to externally assign the indices to the processors. We just want to avoid the collisions explained in my answer to [this comment](https://github.com/open-telemetry/semantic-conventions/pull/1631/files/c62be6444b0c2eb41dde7ab9fd8fcd8c4b4ae7aa#r1866837831).

I think my wording here is misleading:
> I think it's worth documenting that components are responsible for capturing their telemetry and disambiguating it across different instances of the same component.

That's what I actually meant: I didn't want to sell that the SDK itself is responsible, but the component implementations must assign some unique, low-cardinality IDs to their instances.

I'll try to reword it.